README.md

```diff
@@ -51,18 +83,99 @@ Currently available endpoints -
 **Method**: `GET`\
 **Language Code**: IANA standards
 
-(Wiktionary doesn't provide synonyms and antonyms)
+- `http://localhost:8081/v1/dictionary/oxford/{language_code}/pronunciations/{word}` (Oxford-specific)\
+**Method**: `GET`\
+**Language Code**: IANA standards
+
+- `http://localhost:8081/v1/dictionary/oxford/{language_code}/frequency/{word}` (Oxford-specific)\
+**Method**: `GET`\
+**Language Code**: IANA standards
+- `http://localhost:8081/v1/dictionary/oxford/{language_code}/frequency/{word}/{lexicalCategory}` (Oxford-specific)\
+**Method**: `GET`\
+**Language Code**: IANA standards
+
+(Wiktionary only gives definition)
 
 Example endpoints:
 - `http://localhost:8081/v1/dictionary/en/definition/horse`
 - `http://localhost:8081/v1/dictionary/wiktionary/fr/definition/rien`
 - `http://localhost:8081/v1/dictionary/oxford/en/definition/horse`
 
-Tests for these endpoints lie in `share/dictionary/tests`\
-To run them
+### NLP Service
+- `http://localhost:8082/v1/nlp/compromise/tags`\
+**Method**: `POST`\
+**Request Body Format**: `{ sentence: "{sentence_to_be_tagged}" }`
+
+Example endpoints:
+- `http://localhost:8082/v1/nlp/compromise/tags`\
+**Body** `{ sentence: "Hello world! This is a test sentence" }`
+
+### Translation Service
+- `http://localhost:8083/v1/translation/yandex/{source_lang}-{target_lang}`\
+**Method**: `POST`\
+**Request Body Format**: `{ text: "text to be translated"}`\
+**Language Code**: ISO 639-1 (2-lettered language codes)
+
+- `http://localhost:8083/v1/translation/yandex/detect`\
+**Method**: `POST`\
+**Request Body Format**: `{ text: "text for which language is to be detected"}`\
+**Language Code**: ISO 639-1 (2-lettered language codes)
+
+Example endpoints:
+- `http://localhost:8083/v1/translation/yandex/en-de`\
+**Body** `{ text: "This is the text to be translated" }`
+- `http://localhost:8083/v1/translation/yandex/detect`\
+**Body** `{ text: "This is the text to be translated" }`
+
+## Running Tests
+
+Tests for these endpoints lie in -
+- `share/dictionary/tests/` - Tests for dictionary endpoints
+- `share/nlp/tests/` - Tests for nlp endpoints
+- `share/translation/tests/` - Tests for translation endpoints
+
+
+There are 3 kinds of tests for each service -
+- **Unit Tests** - Tests for individual functions
+- **Integration Tests** - Testing the overall functioning of the endpoint (using mock test servers)
+- **Contract Tests** - Testing if the external service provides data with the expected structure
```

Comment:
Regarding the contract tests (and generally) - it would be useful to have some form of error detection and flagging if a needed key does not exist, perhaps when starting a service known to depend on that key.